spatialos/UnrealGDK
# Tests for RPCService

**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
These test are designed to test the interface of the RPCService.

Tests Include:
- Checks that pushing RPC in various configurations returns expected results. 
- Check that the RPCPayloads we push onto the service are used correctly to create corresponding UpdateToSend. 
- Check that if we push RPCs whilst we do not have the entity in the component view that the RPCs are cached and reflect the RPCPayloads we pushed.
- Check that if we push a multi-cast RPC whilst we do not have the entity in the component view that the initial preset is set.
- Check that the Worker_ComponentUpdates we receive have corresponding data to the RPCPayloads we pushed.  

#### Release note
REQUIRED: Add a release note to the `##Unreleased` section of CHANGELOG.md. You can find guidance for writing useful release notes [here](../SpatialGDK/Extras/internal-documentation/how-to-write-good-release-notes.md). Documentation changes are exempt from this requirement.

#### Primary reviewers
@samiwh @aleximprobable
